app/Models/Mship/Account.php

```diff
@@ -975,6 +976,10 @@ public function calculatePasswordExpiry($temporary = false)
      */
     public function setPassword($password, $temporary = false)
     {
+        if (!\Auth::user()) {
+            // user not authenticated, disallow password change in this manner.
+            throw new AuthenticationException();
```

Comment:
This function should still reset a user's password even if they're not logged in (it's up to the controller to implement the logic that ensures the action is authorized, rather than the model - the model simply does what it's told).

Instead, line 985 (of the previous version) should implement an `Auth::check()`, e.g. `if ($save && Auth::check() && \Auth::user()->id === $this->id)`